### DISCUSSION TITLE
Long running TCP connections

### DISCUSSION THREAD

**Discussion by tuxflo** [basecamp/kamal#678] [Q&A]
I just found kamal and it might be the tool that I was looking for our application, which is currently deployed via docker swarm/stack.
One thing we're currently facing is, that our application has some long-running plain TCP streams which should be handled in a zero downtime deployment.
In the documentation I saw the config values `stop_wait_time` and the section with the cord files which states

> then wait for container to become unhealthy.

So my question is: is it possible to configure kamal with a "long-running draining" so that old, active sessions could be drained during a deployment?
How would I approach this, should the application manually write the unhealthy state or modify the cord file?
Also, is there a maximum value for the `stop_wait_time`? Would it be possible to configure `stop_wait_time` for multiple days or even up to two weeks?

**Comment by nickhammond** [comment]
@tuxflo I haven't tried this but I'm pretty sure the Kamal deploy would just hang waiting for the defined `stop_wait_time`. After it issues the stop command on the container you'll see there are docker commands checking to see if that old container is still running. The `stop_wait_time` is just passed as `docker stop -t #{stop_wait_time}` so I doubt there's a maximum, just whatever Docker supports.
